packages/slice-machine/src/features/autoSave/useAutoSave.tsx

```diff
@@ -0,0 +1,114 @@
+import { useCallback, useState, useEffect } from "react";
+import { uniqueId } from "lodash";
+import { toast } from "react-toastify";
+import { Box, Button, Text } from "@prismicio/editor-ui";
+
+type UseAutoSaveArgs = {
+  debounceDelay?: number;
+  errorMessage?: string;
+  retryDelay?: number;
+  retryMessage?: string;
+};
+
+type AutoSaveState = {
+  pendingSave: (() => Promise<unknown>) | null;
+  nextSave: (() => Promise<unknown>) | null;
+};
+
+export type AutoSaveStatusType = "saving" | "saved" | "error";
```

Comment:
I did that because I have a visual component named `AutoSaveStatus` that also refer to the type. So I wanted to be clear about the difference. WDYT?

Good idea for `failed`, I will rename it 🙂

Reply:
That's a good idea, yes!

Reply:
Done